src/main/java/org/miracum/etl/fhirtoomop/interceptors/ClientCredentialsInterceptor.java

```diff
@@ -0,0 +1,90 @@
+package org.miracum.etl.fhirtoomop.interceptors;
+
+import ca.uhn.fhir.rest.client.api.IHttpRequest;
+import ca.uhn.fhir.rest.client.interceptor.BearerTokenAuthInterceptor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+@Slf4j
+public class ClientCredentialsInterceptor extends BearerTokenAuthInterceptor {
+  public static final String GRANT_TYPE = "grant_type";
+  public static final String CLIENT_ID = "client_id";
+  public static final String CLIENT_SECRET = "client_secret";
+  public static final String CLIENT_CREDENTIALS = "client_credentials";
+
+  private SimpleKeycloakAccessToken currentToken;
+  private String tokenUrl;
+  private String clientId;
+  private String clientSecret;
+  private Integer tokenMinValidityBuffer;
+
+  public ClientCredentialsInterceptor(
+      String clientId, String secret, String tokenUrlIn, Integer tokenMinValidityBuffer) {
+    this.clientId = clientId;
+    this.clientSecret = secret;
+    this.tokenUrl = tokenUrlIn;
+    this.tokenMinValidityBuffer = tokenMinValidityBuffer;
+  }
+
+  @Override
+  public void interceptRequest(IHttpRequest theRequest) {
+    log.info("ClientCredentialsIntercepter has intercepted the request");
+    log.info("ClientId: " + clientId);
+    log.info("clientSecret: " + clientSecret);
```

Comment:
Should this be `debug` level? We don't want all these logs every request

Reply:
I had them to help me debug, don't need them now.